script.js

```diff
@@ -1,4 +1,294 @@
+var rollDice = function () {
+  // produces a float between 0 and 6
+  var randomFloat = Math.random() * 6;
+  // take off the decimal
+  var resultInteger = Math.floor(randomFloat) + 1;
+  console.log(`rollDice function, ${resultInteger}`);
+  return resultInteger;
+};
+
+// Player 1 Result Function
+var player1Result = function (player1, player2, userChosenDiceOrder) {
+  var myOutputValue = "";
+  if (currentGameMode == "player1Roll") {
+    var diceRoll1 = rollDice();
+    player1dice1Arr.push(diceRoll1);
+    console.log(`dice roll 1 : ${diceRoll1}`);
+    console.log(`player 1 dice 1 array, ${player1dice1Arr}`);
+
+    var diceRoll2 = rollDice();
+    player1dice2Arr.push(diceRoll2);
+    console.log(`dice roll 2 : ${diceRoll2}`);
+    console.log(`player 1 dice 2 array, ${player1dice2Arr}`);
+
+    //Summary Result Message + Note on how to choose dice order
+    myOutputValue = `Your rolled dice result: <br><br>
+  dice 1 : ${player1dice1Arr} <br><br>
+  dice 2 : ${player1dice2Arr} <br><br>
+  Please choose the order of desired dice result. <br>
+  Type 1 and click enter if you wish for dice 1 to be the first number <br>
+  Type 2 and click enter if you wish for dice 2 to be the first number`;
+
+    currentGameMode = "player1Choice"; //Change game mode to allow player 1 to decide dice order
+    console.log(`diceResult of both dice rolled, ${myOutputValue}`);
+  }
+  // Chosen dice 1 to be first
+  else if (currentGameMode == "player1Choice" && userChosenDiceOrder == 1) {
+    myOutputValue = `${player1} have chose dice 1 to be the first in order, your result is ${player1dice1Arr}${player1dice2Arr} <br><br>
+     ${player2}, you may now begin to roll the dice by clicking enter.`;
+
+    player1Score = `${player1dice1Arr}${player1dice2Arr}`; // Player 1 score
+
+    console.log(`player 1 score, ${player1Score}`);
+    console.log(`user Chosen Dice Order, ${userChosenDiceOrder}`);
+    console.log(`Dice 1 as first number results, ${myOutputValue}`);
+
+    currentGameMode = "player2Roll"; //Change game mode to player 2 turn to roll
+    console.log(
+      `Player 1 turn end, now is player 2 turn. Current mode : ${currentGameMode}`
+    );
+
+    return myOutputValue;
+  } else if (currentGameMode == "player1Choice" && userChosenDiceOrder == 2) {
+    myOutputValue = `${player1} have chose dice 2 to be the first in order, your result is ${player1dice2Arr}${player1dice1Arr} <br><br>
+     ${player2}, you may now begin to roll the dice by clicking enter.`;
+
+    player1Score = `${player1dice2Arr}${player1dice1Arr}`; // Player 1 score
+
+    console.log(`player 1 score, ${player1Score}`);
+    console.log(`user Chosen Dice Order, ${userChosenDiceOrder}`);
+    console.log(`Dice 2 as first number results, ${myOutputValue}`);
+
+    currentGameMode = "player2Roll"; // Change game mode to player 2 turn to roll
+    console.log(
+      `Player 1 turn end, now is player 2 turn. Current mode : ${currentGameMode}`
+    );
```

Comment:
even within `player1Result`, this is fairly repetitive as well. notice how the only difference is really line 78 and line 94. one way i would improve this is to compute the score depending on if the dice order is 1 or 2, then construct your output and switch game mode.

```
  else if (currentGameMode == "player1Choice") {
    if (userChosenDiceOrder == 1) {
      player1Score = `${player1dice1Arr}${player1dice2Arr}`;
    } else {
      player1Score = `${player1dice2Arr}${player1dice1Arr}`;
    }

    myOutputValue = `${player1} have chose dice 1 to be the first in order, your result is ${player1Score} <br><br>
     ${player2}, you may now begin to roll the dice by clicking enter.`;

    currentGameMode = "player2Roll"; //Change game mode to player 2 turn to roll
    return myOutputValue;
  }
```

Reply:
also consider having this part in its own function

Your gameplay logic and code can be broken into three distinct parts: 1. players roll, 2. players select order, 3. compare score. This provides a very nice starting point on how we can split the game into separate functions.

you already have the 3rd one with your `gameOutcome` function, but your players roll and players select order is currently together in one function, and you have one function for every player. this is not scalable esp if we introduce more players. Think of how you can come up with say, a `playerRoll` and `playerSelectOrder` function that works for every player